src/engine/mailprivacy_gnupg.c

```diff
@@ -2830,8 +2830,8 @@ static int mailprivacy_gnupg_init_lock_done = 0;
 #define UNLOCK() LeaveCriticalSection(&encryption_id_hash_lock)
 #endif
 #else
-#define LOCK() do while (0)
-#define UNLOCK() do while (0)
+#define LOCK() {}
+#define UNLOCK() {}
```

Comment:
Did we need to make this change?

Reply:
For some reason 'do while (0)' doesn't compile with our compiler. I would expect that the syntax should be 'do {} while (0)'. So my suspicion is that the empty macro has never been compiled on macOS (or at all) and it's a glitch in the configure script. I just used '{}' since it's the most minimal 'do nothing' C macro (other than perhaps an empty string).

This is caused because configure is turning off the multithreading support because it isn't detecting a needed library. I was initially worried about that, but the only place that the multithreading is used is is in the message encryption/signing which we don't currently use. If we ever implement that functionality then we might need to re-investigate the build setting to get configure to re-enable support.

Both of these issues make me suspect that the 'new' libetpan we're using is slightly different than the old one (even if the source were identical we're obviously compiling it differently). I'm not particularly worried about this change (since I looked into it) but I am a bit worried about any other issues this may expose.

So a possible source of problems is libetpan changes as well as the weird hackiness of my fix. I'm hoping that if we have any issues they surface in the beta. And moving forward we'll at least have more control over the version of libetpan that we use, and if necessary we can change it.

Reply:
Yeah I thought it should've been `do {} while (0)` too, but I guess it doesn't make a difference without any args.

Cool, well let's get this merged in and see how it works :)